hopsworks-common/src/main/java/io/hops/hopsworks/common/elastic/ElasticClient.java

```diff
@@ -0,0 +1,474 @@
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * An Elastic transport client shared by elastic controllers in hopsworks.
+ * An Elastic transport client opens quite a few tcp connection that are kept alive for possibly minutes, and in test
+ * cases for 50-100 parallel ongoing hopsworks elastic based requests it end up with upwards of 2000 tcp connections
+ * used by elastic clients - this can also saturate the amount of tcp connection configured by the elastic node
+ * ending up with a NoNodeAvailableException
+ */
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
+public class ElasticClient {
+  private static final Logger LOG = Logger.getLogger(ElasticClient.class.getName());
+  
+  @EJB
+  private Settings settings;
+  private Client elasticClient = null;
+  
+  @PostConstruct
+  private void initClient() {
+    try {
+      getClient();
+    } catch (ServiceException ex) {
+      LOG.log(Level.SEVERE, "", ex);
+    }
+  }
+  @PreDestroy
+  private void closeClient(){
+    shutdownClient();
+  }
+  
+  /**
+   * Shuts down the client
+   * <p/>
+   */
+  private void shutdownClient() {
+    if (elasticClient != null) {
+      elasticClient.close();
+      elasticClient = null;
+    }
+  }
+  
+  private synchronized Client getClient() throws ServiceException {
+    if (elasticClient == null) {
+      LOG.log(Level.INFO, "creating new elastic client");
+      final org.elasticsearch.common.settings.Settings settings
+        = org.elasticsearch.common.settings.Settings.builder()
+        .put("client.transport.sniff", true) //being able to retrieve other nodes
+        .put("cluster.name", "hops").build();
+      
+      List<String> elasticAddrs = getElasticIpsAsString();
+      TransportClient _client = new PreBuiltTransportClient(settings);
+      for(String addr : elasticAddrs){
+        _client.addTransportAddress(new TransportAddress(new InetSocketAddress(addr, this.settings.getElasticPort())));
+      }
+      elasticClient = _client;
+      
+      Iterator<ThreadPool.Info> tpInfoIt = elasticClient.threadPool().info().iterator();
+      StringBuilder tp = new StringBuilder();
+      while(tpInfoIt.hasNext()) {
+        ThreadPool.Info tpInfo = tpInfoIt.next();
+        switch(tpInfo.getName()) {
+          case ThreadPool.Names.BULK:
+          case ThreadPool.Names.INDEX:
+          case ThreadPool.Names.SEARCH:
+            tp.append("name:").append(tpInfo.getName())
+              .append("(")
+              .append(tpInfo.getThreadPoolType())
+              .append(",")
+              .append(tpInfo.getMin())
+              .append(",")
+              .append(tpInfo.getMax())
+              .append(",")
+              .append(tpInfo.getQueueSize().singles())
+              .append(")");
+        }
+      }
+      LOG.log(Level.INFO, "threadpools {0}", tp);
```

Comment:
@o-alex how much will this log? We could log the theadpool size instead?

Reply:
elastic client creation should not be an event that happens often...